test/unit/spec/util/jobLogViewerSpec.js

```diff
@@ -48,217 +222,493 @@ define(['util/jobLogViewer', 'common/runtime'], (JobLogViewer, Runtime) => {
+            expect(this.node.querySelector('div[data-element="status-line"]')).toBeDefined();
+            this.jobLogViewerInstance.detach();
+            expect(this.node.innerHTML).toBe('');
         });
 
-        it('Should send a bus messages requesting job status information at startup', async (done) => {
-            const viewer = JobLogViewer.make();
-            const jobId = 'testJob1';
+        it('Should send bus messages requesting job status information at startup', async function () {
+            const jobId = 'test_bus_request';
             const arg = {
-                node: hostNode,
+                node: this.node,
                 jobId: jobId,
             };
-            runtimeBus.on('request-job-status', (msg) => {
-                expect(msg).toEqual({ jobId: jobId });
-                viewer.detach();
-                done();
+            await this.jobLogViewerInstance.start(arg).then(() => {
+                return new Promise((resolve) => {
+                    this.runtimeBus.on('request-job-status', (msg) => {
+                        expect(msg).toEqual({ jobId: jobId });
+                        resolve();
+                    });
+                });
             });
-            await viewer.start(arg);
         });
 
-        it('Should react to job status messages', async (done) => {
-            const viewer = JobLogViewer.make();
-            const jobId = 'testJobStatusMsg';
+        it('Should start with all buttons disabled', async function () {
             const arg = {
-                node: hostNode,
-                jobId: jobId,
+                node: this.node,
+                jobId: 'testBtnState',
             };
-            runtimeBus.on('request-job-status', (msg) => {
-                expect(msg).toEqual({ jobId: jobId });
-                runtimeBus.send(
-                    {
-                        jobId: jobId,
-                        jobState: {
-                            status: 'running',
-                        },
-                    },
-                    {
-                        channel: {
+            await this.jobLogViewerInstance.start(arg);
+            const btns = this.node.querySelectorAll('div[data-element="header"] button');
+            btns.forEach((btn) => {
+                expect(btn).toHaveClass('disabled');
+            });
+        });
+
+        describe('initial widget state', () => {
+            describe('should be determining the state with no job object', () => {
+                beforeEach(async function () {
+                    await this.jobLogViewerInstance.start({
+                        node: this.node,
+                        jobId: 'no state set',
+                    });
+                });
+                itHasJobStatus();
+            });
+
+            JobsData.validJobs.forEach((jobState) => {
+                describe(`should create a string for status ${jobState.status}`, () => {
```

Comment:
Valid job state object supplied -- the appropriate job status line will be produced.